pkg/list/values/extract.go

```diff
@@ -0,0 +1,308 @@
+package values
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/cloudposse/atmos/pkg/list/errors"
+	"github.com/cloudposse/atmos/pkg/utils"
+)
+
+// handleSpecialComponent processes special components like settings and metadata.
+func handleSpecialComponent(stack map[string]interface{}, component string) (map[string]interface{}, bool) {
```

Comment:
Not sure that I understand this function name. The `settings` and `metadata` are properties of a component.

Reply:
Maybe `handleComponentProperties`?